x/gov/client/rest/query.go

```diff
@@ -379,10 +377,18 @@ func queryVotesOnProposalHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
 
 		// For inactive proposals we must query the txs directly to get the votes
 		// as they're no longer in state.
+		params := types.NewQueryProposalVotesParams(proposalID, page, limit)
+
 		propStatus := proposal.Status
 		if !(propStatus == types.StatusVotingPeriod || propStatus == types.StatusDepositPeriod) {
 			res, err = gcutils.QueryVotesByTxQuery(cliCtx, params)
 		} else {
+			bz, err := cliCtx.Codec.MarshalJSON(params)
+			if err != nil {
+				rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+				return
+			}
+
 			res, _, err = cliCtx.QueryWithData("custom/gov/votes", bz)
```

Comment:
this error is not checked

Reply:
It's checked in line 395 :arrow_down:

Reply:
Right, not necessarily - I think you're right Fede

Reply:
This looks good